Makefile

```diff
@@ -1,42 +1,107 @@
-# Set the shell to bash always
-SHELL := /bin/bash
+# ====================================================================================
+# Setup Project
+PROJECT_NAME := provider-template
+PROJECT_REPO := github.com/crossplane/$(PROJECT_NAME)
 
-# Options
-ORG_NAME=crossplane
-PROVIDER_NAME=provider-template
+PLATFORMS ?= linux_amd64 linux_arm64
+-include build/makelib/common.mk
 
-build: generate test
-	@CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o ./bin/$(PROVIDER_NAME)-controller cmd/provider/main.go
+# Setup Output
+-include build/makelib/output.mk
 
-image: generate test
-	docker build . -t $(ORG_NAME)/$(PROVIDER_NAME):latest -f cluster/Dockerfile
+# Setup Go
+NPROCS ?= 1
+GO_TEST_PARALLEL := $(shell echo $$(( $(NPROCS) / 2 )))
+GO_STATIC_PACKAGES = $(GO_PROJECT)/cmd/provider
+GO_LDFLAGS += -X $(GO_PROJECT)/internal/version.Version=$(VERSION)
+GO_SUBDIRS += cmd internal apis
+GO111MODULE = on
+-include build/makelib/golang.mk
 
-image-push:
-	docker push $(ORG_NAME)/$(PROVIDER_NAME):latest
+# Setup Kubernetes tools
+-include build/makelib/k8s_tools.mk
 
-run: generate
-	kubectl apply -f package/crds/ -R
-	go run cmd/provider/main.go -d
+# Setup Images
+DOCKER_REGISTRY ?= crossplane
```

Comment:
We have used this so far. Do you think we should keep using that org? I can see us pushing `crossplane-contrib` images to a `crossplane-contrib` org.

Reply:
I'd prefer to mostly propagate what we do today in this PR and make changes in future PRs. So let's keep this for now.